### ISSUE TITLE
[FR] Show remaining time

### ISSUE THREAD

**Issue by MLiv79** [bigtreetech/BIGTREETECH-TouchScreenFirmware#1891]
Would be possible to add the remaining time during a print?

**Comment by oldman4U** [comment]
Please see PR #1410.

**Comment by kisslorand** [comment]
Yes, it's possible if you know the total time needed for the print job.
That would be possible by analyzing the gCode before the print job. Not an impossible task but it would be overkill for the TFTs MCU with little benefit.
It would take some time to analyze it and than people will start tickets that it takes too long for a print to start...

**Comment by oldman4U** [comment]
Or it is part of the gCode. But this only works as long as the speed is not changed.

**Comment by MLiv79** [comment]
> Yes, it's possible if you know the total time needed for the print job.
> That would be possible by analyzing the gCode before the print job. Not an impossible task but it would be overkill for the TFTs MCU with little benefit.
> It would take some time to analyze it and than people will start tickets that it takes too long for a print to start...

Marlin standard do this without any issue. Progress bar, Elapsed Time and remaining time. These are written by the slicer (prusa for example) in the gcode. Using touch mode no remaining time with the same gcode that with Marlin mode show all the data written above, included remaining time.

**Comment by oldman4U** [comment]
Have you tried the PR I mentioned?

**Comment by kisslorand** [comment]
Where should it be displayed? I mean exactly where in the print menu? There is not really any extra space left.

**Comment by oldman4U** [comment]
Using the PR I mentioned above the remaining time is shown INSTEAD of the elapsed time. The final implementation should show the elapsed and remaining time alternating like Flow and Speed is shown today.

**Comment by oldman4U** [comment]
@MLiv79 Please test it and let us know.

**Comment by MLiv79** [comment]
> Using the PR I mentioned above the remaining time is shown INSTEAD of the elapsed time. The final implementation should show the elapsed and remaining time alternating like Flow and Speed is shown today.

Exactly in this way

**Comment by kisslorand** [comment]
How would you know which is which?

**Comment by oldman4U** [comment]
Using different Icons and maybe even Text. Like Speed and Flow.

MLiv79 and, have you tested the PR in the meantime?

**Comment by MLiv79** [comment]
> Using different Icons and maybe even Text. Like Speed and Flow.
> 
> MLiv79 and, have you tested the PR in the meantime?

I'm printing a thing sliced with IM now so I've not looked at it again but the first time I've printed i used prusa and on prusa/superslicer I've m73 enabled (even in cura now is possible) and see only elapsed time. But I've to look at it again. If would be possible in next release also a number at the end of the progress bar would be useful instead of the bar alone. Thanks @oldman4U

**Comment by oldman4U** [comment]
You have to use #1410, otherwise you will not see the remaining time.

**Comment by MLiv79** [comment]
> You have to use #1410, otherwise you will not see the remaining time.

Could you elaborate a little bit more? Because I've m73 p r enabled and i have it in the gcode but on tft i still see the elapsed time

**Comment by oldman4U** [comment]
Which firmware of the TFT ate you using?

**Comment by MLiv79** [comment]
> Which firmware of the TFT ate you using?

Latest .27 compiled by myself (but didn't work even with the ones already compiled included)

**Comment by oldman4U** [comment]
But I asked you to use #1410 and not latest v27...

V27 does not #1410 does show remaining time.

**Comment by MLiv79** [comment]
> But I asked you to use #1410 and not latest v27...
> 
> V27 does not #1410 does show remaining time.

I compiled by myself master.zip release, is not included in there?

**Comment by oldman4U** [comment]
No it is not.

**Comment by MLiv79** [comment]
> No it is not.

in which zip is included? Developer?

**Comment by oldman4U** [comment]
When you go to Pull Request you will find 1410 and there you can download the firmware like normal.

**Comment by MLiv79** [comment]
> When you go to Pull Request you will find 1410 and there you can download the firmware like normal.

This should include all last commit included yours, right?
![image](https://user-images.githubusercontent.com/70020275/116824368-1ae53e80-ab8a-11eb-914b-bf7d70144493.png)

**Comment by oldman4U** [comment]
It is based on an older version of the firmware, but you can see if it does what you want: https://github.com/belese/BIGTREETECH-TouchScreenFirmware.git

**Comment by sarvenn** [comment]
> It is based on an older version of the firmware, but you can see if it does what you want: https://github.com/belese/BIGTREETECH-TouchScreenFirmware.git

The one above or this? https://github.com/belese/BIGTREETECH-TouchScreenFirmware/tree/M77Support

**Comment by oldman4U** [comment]
Should be the same

**Comment by oldman4U** [comment]
Any news?

**Comment by MLiv79** [comment]
> Any news?

It worked for me when i tested it but it is not included in the master

**Comment by oldman4U** [comment]
Ok.

@sarvenn?

**Comment by oldman4U** [comment]
There is a new PR available related to the requested function. Please test and let us know. Thank you

https://github.com/digant73/BIGTREETECH-TouchScreenFirmware.git

**Comment by sarvenn** [comment]
> Ok.
> 
> @sarvenn?

Hi @oldman4U, I am away from home for a couple of weeks due to my on-going projects at my work. I will try that pr asap. But we may merge it now and follow the bug reports as well. It's a matter of choice and I respect both ways.

**Comment by Hitamku** [comment]
> There is a new PR available related to the requested function. Please test and let us know. Thank you
> 
> https://github.com/digant73/BIGTREETECH-TouchScreenFirmware.git

Hai, remaining time not showing. I use Biqu B1. What did i miss?

**Comment by oldman4U** [comment]
Make sure your slicer adds the needed information to the gCode.

**Comment by Hitamku** [comment]
> Make sure your slicer adds the needed information to the gCode.

I've tried activating m73 in marlin, m117 in config, display progess on lcd in Cura. But the remaining time still doesn't appear

**Comment by oldman4U** [comment]
Attach the gCode file.

**Comment by Hitamku** [comment]
I did print test and also did test using gCode test

; FLAVOR: Marlin
M117 Time Left 0h01m40s
G4 P10000
M73 P10
M117 Time Left 0h01m30s
G4 P30000
M73 P40
M117 Time Left 0h01m00s
G4 P30000
M73 P70
M117 Time Left 0h00m30s
G4 P30000
M73 P100
M117 Time Left -1h59m59s

the result is the same. Does it only work in Mks?
